compressors/baseline_compressors.py

```diff
@@ -0,0 +1,73 @@
+"""
+Contains some elementary baseline compressors
+1. Fixed bit width compressor 
+"""
+
+import numpy as np
+from core.data_transformer import (
+    BitsToBitstringTransformer,
+    BitstringToBitsTransformer,
+    BitstringToUintTransformer,
+    CascadeTransformer,
+    LookupTableTransformer,
+    UintToBitstringTransformer,
+)
+from core.data_compressor import DataCompressor
+
+
+class FixedBitwidthCompressor(DataCompressor):
+    """
+    Encodes each symbol of the data to a fixed bitwidth
+    Example: data = [A,B,B,A,C]
+    Encoding ~ ['00',`01`, `01`, `00`, `10`] -> converted to bits
+    See baseline_compressors_tests.py for usage
+    """
+
+    def set_encoder_decoder_params(self, data_stream):
+        """
+        Each compressor consists of encoder_transform + decoder_transform
+        In this case we need to know the input alphabet size to define the transforms
+
+        We first get the alphabet size and then set the encoder and decoder transforms
+        in this function
+        """
+        # set bit width
+        alphabet = data_stream.get_alphabet()
+        self.bit_width = self._get_bit_width(len(alphabet))
+
+        # set encoder/decoder lookup tables
+        self.encoder_lookup_table = {}
+        self.decoder_lookup_table = {}
+
+        for id, a in enumerate(alphabet):
+            self.encoder_lookup_table[a] = id
+            self.decoder_lookup_table[id] = a
+
+        # create encoder and decoder transforms
+        # The encoder transformer is represented as a cascade:
+        # Example:
+        # data = [A,B,B,A,C]
+        # Transform-1: Lookuptable to get he alphabet index {A: 0, B: 1, C: 2}
+        # Transform-2: Uint to bitstring with a fixed width (2 for example) [0 -> 00, 1 -> 01, 2 -> 10]
+        # Transform-3: Bitstring to bits: [000 -> 0,0,0]
+        self.encoder_transform = CascadeTransformer(
+            [
+                LookupTableTransformer(self.encoder_lookup_table),
+                UintToBitstringTransformer(bit_width=self.bit_width),
+                BitstringToBitsTransformer(),
```

Comment:
I am unclear on why you need BitstringToBitsTransformer? Is it similar to converting the block based codes to streaming codes? In that case, should we think of a general transform class going to-and-fro from "block-based" codes to "streamable" codes, e.g. as @shubhamchandak94 suggested for DataStream?

Reply:
@tpulkit, this is the recommended way of leaving comments. Typically we should leave the PR unmerged if there are any un-resolved comments!

Reply:
The final output of any encoder has to be a `BitsDataStream`, to be a valid encoder and to help with figuring out codelength etc. 

See the following snippet in `data_compressors.py`
```

    def encode(self, data_stream: DataStream):
        """
        The core encode function of the compressor
        """

        # set the parameters of the encoder/decoder using the data_stream
        self.set_encoder_decoder_params(data_stream)

        # perform the encoding
        output_bits_stream = self.encoder_transform.transform(data_stream)

        # the final output of encoder needs to be a stream of bits
        assert isinstance(output_bits_stream, BitsDataStream)
```

Reply:
This case, typically we will build a lookup take which will output a bitstring, so the transform just converts the bitstring to bits